### ISSUE TITLE
Allow to override keycloak resource tag for fixing issue with browser caching

### ISSUE THREAD

**Issue by jowko** [keycloak/keycloak#10123]
### Description

Allow to modify resource tag from MigrationModelEntity to a new value.

### Motivation

Keycloak resources url contains a special generated tag (yeqd3):
https://hostname/auth/resources/yeqd3/login/MyTheme/css/common.css
Because of it, on each keycloak version update, this tag is modified and browsers will drop old cache and load new files which could be modified. This works fine if we want to use standard keycloak theme. But if we add our own theme with our company branding to it, them such theme is cached too until we upgrade to a new version of keycloak (each keycloak version change generates a new tag, see MigrationModelAdapter). Then changes which we make to a theme are not visible for some of our customers, because themes were cached. 

To resolve this issue we would need to either disable caching on production or change resource tag to a different value when we make an update to keycloak theme (we release own keycloak docker image which extends keycloak).
Currently we are running SQL script which modifies this table but it depends on internal implementation details of keycloak schema:
```
UPDATE  keycloak.MIGRATION_MODEL set id='$RESOURCE_TAG' where version in (select max(version) from keycloak.MIGRATION_MODEL);
```

Also see:
https://medium.com/@hokumski/how-to-drop-keycloak-theme-cache-without-clearing-browser-data-aebb499ae83b

### Details

We could provide some way to change this via REST API or environment variable.

**Comment by Syn-Tylor** [comment]
Running into this same situation using redhat-sso 7.4 which uses keycloak. Following the second half of that medium.com article above; we can update that MIGRATION_MODEL table directly and get the new resources. While working locally we can simply modify that config (reference in article) and set the  <staticMaxAge> which results in it updating as you work (great).
Though, for our CI/CD this adds a manual step for deployment which we need to avoid.

**Comment by raehalme** [comment]
Would it make sense to allow overriding the resource tag with an environment variable? 

Whenever you modify your custom theme, you will also build a new custom Docker image. Then you can use a different value for the resource tag regardless of the current Keycloak version.

**Comment by jowko** [comment]
@raehalme Yes it would make sense. It is one of the proposal described in task description.

**Comment by raehalme** [comment]
@jowko Sorry I missed the two last words of the description ;)

I'd definitely prefer using environment variables instead of having to update the value separately using a REST API endpoint. This way the value can be directly dependent of the deployment.

**Comment by stianst** [comment]
I'd say a better option would be to allow a theme to define its own version.

When generating the resource tag at startup we can check Keycloak version as well as theme versions, and create a new resource tag if any has changed.

An even better option may be to have a resource tag generated for each theme individually.

**Comment by Lunkwill-Fook** [comment]
Is there any workaround for this except with a SQL command?
